lua/powerups/client/curse.moon

```diff
@@ -67,13 +122,23 @@ timer.Create "CFC_Powerups-Curse-EmitterThink", EMITTER_INTERVAL, 0, ->
         for _ = 1, EMITTER_AMOUNT
             pos = centerPos + Vector(math.Rand(-spreadX, spreadX), math.Rand(-spreadY, spreadY), math.Rand(-spreadZ, spreadZ))
             dir = AngleRand!\Forward!
-            colorIntensity = math.Rand 0, EMMITTER_COLOR_INTENSITY
-            with part = emitter\Add EMITTER_MATERIAL, pos
-                \SetStartSize EMITTER_START_SIZE
-                \SetEndSize EMITTER_END_SIZE
+
+            weight = math.random 1, emitterTotalWeight
+            option = nil
+            for i, total in ipairs emitterWeightTotals
+                if weight <= total
+                    option = EMITTER_OPTIONS[i]
+                    break
+
+            color = option.color
+            intensity = math.Rand option.colorIntensityMin, option.colorIntensityMax
+
+            with part = emitter\Add option.mat, pos
+                \SetStartSize option.startSize
+                \SetEndSize option.endSize
                 \SetDieTime EMITTER_LIFE
                 \SetGravity EMITTER_GRAVITY
-                \SetColor colorIntensity / 2, 0, colorIntensity
+                \SetColor color.r * intensity, color.g * intensity, color.b * intensity
                 \SetVelocity dir * math.Rand EMITTER_SPEED_MIN, EMITTER_SPEED_MAX
 
```

Comment:
yours is fine, i think this is neat though
```moon
intensity = math.Rand option.colorIntensityMin, option.colorIntensityMax
color = [c * intensity for c in *option.color\Unpack!]

with part = emitter\Add option.mat, pos
                \SetStartSize option.startSize
                \SetEndSize option.endSize
                \SetDieTime EMITTER_LIFE
                \SetGravity EMITTER_GRAVITY
                \SetColor unpack color
                \SetVelocity dir * math.Rand EMITTER_SPEED_MIN, EMITTER_SPEED_MAX
```

Reply:
Neat, but not a good application of moon list comprehensions tbh. It's roughly the same amount of chars, but now as a list comprehension the reader has to decipher rather than a simple multiplication being repeated three times with a different color channel each.

It's also *definitely* worse in performance as it uses a for loop, two unpacks, ipairs, creates a new table, and multiplies the unused alpha channel.